### PR TITLE
445: Covered the NoninterceptableInterface case by plugin inspection …

### PR DESCRIPTION
**Pull request by VitaliyBoyko**
…and updated create a plugin action

Now creating a plugin for a class that implements NoninterceptableInterface  causes a warning:
image

Also removed such a possibility from the context menu:
image

And provided a bit of cleanup :)

**Fixed Issues (if relevant)**
1. Fixes magento/magento2-phpstorm-plugin#445

**Contribution checklist** (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with integration/functional tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,16 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0).
 
 ## 3.0.4
 
+### Fixed
+
+- Overriding the interface generates invalid php code
+- Overriding a template from the base area
+- Disabled an ability to create a plugin for a class that implements `NoninterceptableInterface`
```
Comment (eduard13) [inline CHANGELOG.md:13]:
```suggestion
- Overriding the interface that generates invalid php code
- Overriding a template from the base area
- Disabled the ability to create a plugin for a class that implements `\Magento\Framework\ObjectManager\NoninterceptableInterface`
```

File: resources/magento2/inspection.properties [hunk 2]
```diff
@@ -8,6 +8,7 @@ inspection.graphql.resolver.fix.title=Select one of the following interface
 inspection.graphql.schema.resolver.fix.family=Create GraphQL Resolver
 inspection.plugin.error.nonPublicMethod=You can't declare a plugin for a not public method.
 inspection.plugin.error.finalClass=You can't declare a plugin for a final class.
+inspection.plugin.error.noninterceptableInterface=You can't declare a plugin for a class implements Magento\\Framework\\ObjectManager\\NoninterceptableInterface.
```
Comment (eduard13) [inline resources/magento2/inspection.properties:11]:
```suggestion
inspection.plugin.error.noninterceptableInterface=You can't declare a plugin for a class that implements Magento\\Framework\\ObjectManager\\NoninterceptableInterface.
```

**Review by eduard13 — CHANGES_REQUESTED** [review]
Good job 👍 and works great.
I have just couple of suggestions.

**Review by eduard13 — APPROVED** [review]

**Comment by VitaliyBoyko** [comment]
Thanks!

**Pull request merged**